eng/tools/npm-helpers.ps1

```diff
@@ -0,0 +1,162 @@
+
+function GetNpmTagVersions($packageName)
+{
+  try
+  {
+    $existingVersion = Invoke-RestMethod -Method GET -Uri "http://registry.npmjs.com/${PackageName}"
+    $latest = ($existingVersion."dist-tags").latest
+    $next = ($existingVersion."dist-tags").next
+    Write-Host "Latest version: $latest"
+    Write-Host "Next version: $next"
+    if ($latest -eq $null) {
+      Write-Host "'latest' tag is not present in npm registry for package $packageName"
+    }
+    if ($next -eq $null) {
+      Write-Host "'Next' tag is not present in npm registry for package $packageName"
+    }
+
+    return New-Object PSObject -Property @{
+      latest = $latest
+      next = $next
+    }
+  }
+  catch
+  {
+    return $null
+  }
+}
+
+function GetNpmPackageVersions ($packageName)
+{
+  try
+  {
+    Write-Host "Checking versions present on npm for package $packageName"
+    $existingVersion = Invoke-RestMethod -Method GET -Uri "http://registry.npmjs.com/${packageName}"
+    return ($existingVersion.versions | Get-Member -MemberType NoteProperty).Name
+  }
+  catch
+  {
+    return $null
+  }
+}
+
+function FindRecentPackageVersion($packageName)
+{
+  $versions = (GetNpmPackageVersions -packageName $packageName) | ? {$_ -NotMatch "alpha|dev"}  
+  if ($versions -ne $null -and $versions.Count -gt 0)
+  {
+    $versions = [AzureEngSemanticVersion]::SortVersionStrings($versions)
+    $highestNpmVersion = $versions[0]
+    Write-Host "Recent version uploaded to NPM: $highestNpmVersion"
+    return $highestNpmVersion
+  }
+  
+  return $null
+}
+
+function GetNewNpmTags($packageName, $packageVersion)
+{
+  $newVersion = [AzureEngSemanticVersion]::ParseVersionString($packageVersion)
+  Write-Host "Package name: $packageName"
+  Write-Host "Package version: $packageVersion"
+  Write-Host "Find latest and next versions in npm registry for package"
+
+  <#
+  1. Get latest GA and latest preview from npm
+  2. Check new version to be published to find is it GA or preview
+  3. If new Version is GA:
+    a. If higher than current latest, or first GA then set LATEST tag
+  4. If new version is preview and higher than current higher version in npm:
+    a. Set LATEST if package has never GA released
+    b. Set NEXT tag
+  #>
+  $npmVersionInfo = GetNpmTagVersions -packageName $packageName
+  if ($npmVersionInfo -eq $null)
+  {
+    # Version info object should not be null even if package is not present in npm
+    Write-Error "Failed to get version info from NPM registry."
+  }
+  $latestVersion = [AzureEngSemanticVersion]::ParseVersionString($npmVersionInfo.latest)
+  $setLatest = $false
+  $setNext = $false
+  # Set Latest tag if new version is higher than current GA or if package has never GA released before
+  if ((!$newVersion.IsPreRelease) -and ($latestVersion -eq $null -or $newVersion.CompareTo($latestVersion) -eq 1)) {
+    $setLatest = $true
+  }
+
+  if ($newVersion.PrereleaseLabel -eq "preview" -or $newVersion.PrereleaseLabel -eq "beta")
```

Comment:
Are we sure that we only want to set the `next` tag if we are releasing a beta? For example, I just released `@azure/ai-form-recognizer@3.1.0`, and I tagged it both "latest" and "next," reason being that it is newer than the newest beta, so it didn't make sense to me that someone who was using the `next` tag wouldn't get that release, while someone using the `latest` tag would, so I believe the right thing to do in those cases is to set _both_ "latest" and "next".

In fact, I'm not sure there's any situation where we would set "latest" but not "next" as well. Any time we promote a package from beta to GA, it becomes both "next" and "latest," and beta releases only become "next."

Reply:
As per the earlier discussion with Ramya, next tag should not be set for GA package. It should only be set for preview or beta versions. so latest and next will be set for a version only if that package has never GA released and if current version is preview. So my understanding is to set next to next possible GA version ( which is current preview or GA), not to current GA version.

Reply:
@witemple-msft Thats what we do today. See the guidelines at https://github.com/Azure/azure-sdk/blob/e1c52031fc395a31202472db7e72e8bed8e1bffa/docs/policies/releases.md#javascript

The idea behind not having the `next` tag on a GA version is due to the perception that `next` points to an unstable version. I understand that if one was using the `next` tag in their package.json, they might suddenly find their `npm install` step failing. But I would expect users to rather pin their preview dependencies instead of use `next` as there is no guarantee that the next `next` tag will be non breaking. Also `npm` makes no guarantee that any tag other than `latest` would be available at any given point.

We can revisit this discussion, but not in this PR. You can create an issue in https://github.com/Azure/azure-sdk where our guidelines live if you want to propose a change